### ISSUE TITLE
[A11y][iOS] unexpected hittest order when using voiceover

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#177025]
### Steps to reproduce

1. turn on voiceover
2. open the dialog by clicking the button
3. tap anywhere inside the dialog except the two text fields

### Expected results

a11y focus should stay where they are

### Actual results

a11y focus shifts to dismiss modal barrier. This is due to the the hittest continue to the next child even through the pointer should be absorb when hittest hit in bound
https://github.com/flutter/flutter/blob/56ce192dc679a059c1f5c566e373669c96157123/engine/src/flutter/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L548

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
  WidgetsFlutterBinding.ensureInitialized().ensureSemantics();
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(title: 'Semantics Issue', home: MyHomePage());
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  void _onPressed() {
    showDialog(
      context: context,
      barrierColor: Colors.grey.shade400,
      builder: (context) {
        return _dialogContent;
      },
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Stack(
          children: <Widget>[
            Semantics(
              container: true,
              child: FilledButton(
                onPressed: _onPressed,
                child: const Text('Show Dialog'),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget get _dialogContent => Center(
    child: Container(
      margin: const EdgeInsets.all(50),
      child: const Material(
        child: SizedBox(
          height: 250,
          width: 250,
          child: Padding(
            padding: EdgeInsets.all(20),
            child: Column(
              children: [
                TextField(
                  decoration: InputDecoration(
                    border: OutlineInputBorder(),
                    labelText: 'Form field one',
                  ),
                ),
                SizedBox(height: 40),
                TextField(
                  decoration: InputDecoration(
                    border: OutlineInputBorder(),
                    labelText: 'Form field two',
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    ),
  );
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by LongCatIsLooong** [comment]
From triage: for context https://github.com/flutter/flutter/commit/13c33441ef7dec14b23748789ed8441181a27880 is the commit that added the logic.
